From f5d1090b0b4326fb705d5afb62001c29958087f7 Mon Sep 17 00:00:00 2001
From: NLObP <38990417+NL0bP@users.noreply.github.com>
Date: Thu, 8 Aug 2024 16:56:07 +0300
Subject: [PATCH] [Game][Ai] Fixed support distance calculation: (#1064)

- They incorrectly calculated the distance to themselves. We calculate the NPC that the player attacked.
---
 AAEmu.Game/Models/Game/AI/v2/Framework/Behavior.cs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/AAEmu.Game/Models/Game/AI/v2/Framework/Behavior.cs b/AAEmu.Game/Models/Game/AI/v2/Framework/Behavior.cs
index bd5110dfa7..b2b5278075 100644
--- a/AAEmu.Game/Models/Game/AI/v2/Framework/Behavior.cs
+++ b/AAEmu.Game/Models/Game/AI/v2/Framework/Behavior.cs
@@ -359,7 +359,7 @@ public void UpdateAggroHelp(Unit abuser, int radius = 200)
 
         foreach (var npc in npcs
                      .Where(npc => !npc.IsInBattle && npc.Template.AcceptAggroLink)
-                     .Where(npc => npc.GetDistanceTo(npc.Ai.Owner) <= npc.Template.AggroLinkHelpDist))
+                     .Where(npc => npc.GetDistanceTo(Ai.Owner) <= npc.Template.AggroLinkHelpDist))
         {
             if (npc.Template.Aggression && npc.Template.AggroLinkSpecialRuleId == AggroLinkSpecialRuleKind.None)
             {
